app/src/controllers/reservation/services/ReservationService.ts

```diff
@@ -167,6 +213,30 @@ const ReservationService: ReservationServiceInterface = {
       console.error('Invalid date, earlier than today')
       throw new InvalidParamsError()
     }
+    const filteredDates = filterAndGetWantedDate(reservationDate)
+
+    const reservationDatesAndDuration = await ReservationRepository.fetchReservationsByDSS(
+      shopId, filteredDates.startDate,
+      filteredDates.endDate, stylistId,
+    )
+
+    const shopSchedule = await ShopRepository.fetchShopSchedule(shopId)
+
+    const menuDuration = await ReservationRepository.fetchMenuDuration(menuId)
```

Comment:
maybe use menu repository for this

Reply:
you can actually just change the menu exists of menu repository to just fetch the duration itself then use it as a check if the menu exists. it's being done on line 201